owncloudApp/src/main/java/com/owncloud/android/ui/helpers/FileOperationsHelper.java

```diff
@@ -442,9 +440,9 @@ private void shareLink(String link) {
                     )
             );
         }
-
-        String[] packagesToExclude = new String[]{mFileActivity.getPackageName()};
-        DialogFragment chooserDialog = ShareLinkToDialog.newInstance(intentToShareLink, packagesToExclude);
-        chooserDialog.show(mFileActivity.getSupportFragmentManager(), FTAG_CHOOSER_DIALOG);
+         ShareLinkToShareSheet shareLinkToShareSheet = ShareLinkToShareSheet.Companion.getInstance(intentToShareLink, mFileActivity, componentName -> {
+                    return componentName.getPackageName().equals(mFileActivity.getPackageName());
+                });
+        shareLinkToShareSheet.show();
```

Comment:
I think you overcomplicated this. If you check this [documentation](https://developer.android.com/training/sharing/send#send-text-content), you will see that you could simply call

`        mFileActivity.startActivity(Intent.createChooser(intentToShareLink, null));
`

And it will open the sharesheet as expected. In newer versions you will see a copy to clipboard if you are sharing text etc

There are two things to take care when we are sharing text:

1- Do we need to show ownCloud as target? I don't think so
2- Do we need to add some extra text to the share link? For example: "demo @owncloud.com shared this link(http:server/link) with you" I think it would be enough to copy the link.

Opinions? @michaelstingl @jesmrec